hashable.cabal

```diff
@@ -61,6 +61,15 @@ flag random-initial-seed
   manual:      True
   default:     False
 
+
+flag sse2
+  description: Do we want to assume that a target supports SSE 2?
```

Comment:
`Use SSE2 instruction set. Available on virtually any x86 CPU made since 2005.` i.e. don't ask, tell.

For `sse4` write something similar.

Reply:
FWIW, the latest `text` version assumes that SSE2 is always available on x64, see e.g. https://hackage.haskell.org/package/text-1.2.5.0/src/cbits/cbits.c

```c
#if defined(__x86_64__)
  /* All the intrinsics used here are from SSE2,
   * so every x86_64 CPU supports them.
   */
```

I actually don't care if i386 is slow. I don't have machines to test it anyway, so being simple on these is a virtue.

Reply:
So I take from this that sse2 can be enabled without flag, and sse4 probably can be enabled by default at this point in time. 

(I still need to do this btw).

Reply:
sse2 on x64. On i386 i'd rather *not* have conditional branches at all, as they are hard to test nowadays.